buildkite/bazelci.py

```diff
@@ -2960,57 +2917,52 @@ def print_incompatible_flags_info_box_step(incompatible_flags_map):
 
 def fetch_incompatible_flags():
     """
-    Return a list of incompatible flags to be tested in downstream with the current release Bazel
+    Return a list of incompatible flags to be tested. The key is the flag name and the value is its Github URL.
     """
-    incompatible_flags = {}
-
-    # If INCOMPATIBLE_FLAGS environment variable is set, we get incompatible flags from it.
-    if "INCOMPATIBLE_FLAGS" in os.environ:
-        for flag in os.environ["INCOMPATIBLE_FLAGS"].split():
-            # We are not able to get the github link for this flag from INCOMPATIBLE_FLAGS,
-            # so just assign the url to empty string.
-            incompatible_flags[flag] = ""
-        return incompatible_flags
-
     output = subprocess.check_output(
         [
+            # Query for open issues with "incompatible-change" and "migration-ready" label.
             "curl",
-            "https://api.github.com/search/issues?per_page=100&q=repo:bazelbuild/bazel+label:incompatible-change+state:open",
+            "https://api.github.com/search/issues?per_page=100&q=repo:bazelbuild/bazel+label:incompatible-change+label:migration-ready+state:open",
         ]
     ).decode("utf-8")
     issue_info = json.loads(output)
 
+    FLAG_PATTERN = re.compile(r"^--[a-z][a-z0-9_]*$")
+    incompatible_flags = {}
     for issue in issue_info["items"]:
-        # Every incompatible flags issue should start with "<incompatible flag name (without --)>:"
         name = "--" + issue["title"].split(":")[0]
         url = issue["html_url"]
-        if name.startswith("--incompatible_"):
+        if FLAG_PATTERN.match(name):
```

Comment:
Have we dropped the --incompatible_ prefix requirement?

Reply:
Yes, but it's still a recommendation. See https://github.com/bazelbuild/bazel/pull/16195

@meisterT mentioned that people are also shipping some incompatible change under flags that don't start with `--incompatible_`. And if you consider Bzlmod is an incompatible change, then`--enable_bzlmod` also doesn't fit.